### ISSUE TITLE
Node registration seems to have contention

### ISSUE THREAD

**Issue by lpellegr** [ow2-proactive/scheduling#2359]
It was noticed while doing some micro-benchmarks on EC2 that registering many PA nodes at the same time on a single scheduler instance (having 4 CPUs) is _slow_.  

For instance, adding _1000_ ProActive nodes takes about 14 minutes and 30 seconds based on client sides output provided below. First 640 registrations are pretty fast (about 1 minute), however next ones are much slower:

```
[1] 17:28:56 [SUCCESS] 52.28.82.232
0
[1] 17:29:04 [SUCCESS] 52.28.82.232
38
[1] 17:29:11 [SUCCESS] 52.28.82.232
80
[1] 17:29:18 [SUCCESS] 52.28.82.232
147
[1] 17:29:26 [SUCCESS] 52.28.82.232
241
[1] 17:29:33 [SUCCESS] 52.28.82.232
399
[1] 17:29:41 [SUCCESS] 52.28.82.232
597
[1] 17:29:48 [SUCCESS] 52.28.82.232
620
...
[1] 17:32:30 [SUCCESS] 52.28.82.232
980
...
[1] 17:34:05 [SUCCESS] 52.28.82.232
982
...
[1] 17:38:49 [SUCCESS] 52.28.82.232
989
...
[1] 17:43:26 [SUCCESS] 52.28.82.232
1000
```

Looking at scheduler logs reveals that nothing else seems performed during the freeze periods that were noticed previously (hours are not synchronized in logs since machines do not use the same time zone):

```
[2015-10-16 11:29:39,087 INFO            o.o.p.r.n.NodeSource] [Default] new node available : pnp://ip-172-31-29-170.eu-central-1.compute.internal:40947/ip-172-31-29-170_eu-central-1_compute_internal_4834
[2015-10-16 11:31:56,442 INFO            o.o.p.r.n.NodeSource] Looking up the node pnp://ip-172-31-29-163.eu-central-1.compute.internal:45602/ip-172-31-29-163_eu-central-1_compute_internal_3728 with 30000 ms timeout
```

I suspect some contention on internal data structures (or maybe DB) due to the number of requests (PA nodes registrations) to handle. 

Adding a sleep period (still using the same hardware) to avoid a burst of requests on scheduler side seems to improve a lot registration time. For instance, sending 75 registrations to the scheduler each second allows to insert 1500 ProActive nodes in 6 minutes 44 seconds.

It seems to confirm that the issue is related to thread contention or DB contention. However, more investigations are required to know which internal data structures are affected and check if something can be improved.

**Comment by lpellegr** [comment]
As discussed with Fabien, the problem could also be linked with topology discovery.

**Comment by fviale** [comment]
Each time a node is added a topology calculation is done. This implies, for each new node, the creation of an active object on it, which will ping sequentially all other nodes.

Two remarks:

1) The NodeSet ping implementation is wrongly designed, it pings sequentially each node one by one, where by using asychronous method calls and futures, it could ping all nodes at once.

2) node topology could be disabled by default as it is not widely used. MultiNode selection in most cases rely on other strategies than "bestProximity".

**Comment by fviale** [comment]
The customer reports that the fix of #2373 really removed the observed contention. So I close this issue
